lib/engine/game/g_1848/step/buy_train.rb

```diff
@@ -32,14 +37,36 @@ def buyable_trains(entity)
             # Cannot buy 2E if one is already owned
             trains_to_buy = super
             trains_to_buy = trains_to_buy.select(&:from_depot?) unless @game.can_buy_trains
-            trains_to_buy = trains_to_buy.reject { |t| t.name == '2E' } if owns_2e?(entity)
-            trains_to_buy
+            if owns_2e?(entity)
+              trains_to_buy = trains_to_buy.reject { |t| t.name == '2E' }
+            elsif trains_to_buy.none? { |t| t.name == '2E' } && can_buy_2e?(entity)
+              trains_to_buy << ghan_train
+            end
+            trains_to_buy.uniq
           end
 
           def owns_2e?(entity)
             entity.trains.any? { |t| t.name == '2E' }
           end
 
+          def can_buy_2e?(entity)
```

Comment:
If this method is going to only apply to the discount, maybe change the name to can_buy_2e_with_discount?(entity)?

Reply:
it doesn't, i'll need to use it for a bug i found. When a company is at train limit, it can buy a 2e currently. room?(entity) needs to check when at limit if the corp can buy the 2e, and will rely on this method